contracts/modules/dispute/policies/UMA/ArbitrationPolicyUMA.sol

```diff
@@ -0,0 +1,272 @@
+        $.maxBonds[token] = maxBond;
+
+        emit MaxBondSet(token, maxBond);
+    }
+
+    /// @notice Executes custom logic on raising dispute
+    /// @dev Enforced to be only callable by the DisputeModule
+    /// @param caller Address of the caller
+    /// @param data The arbitrary data used to raise the dispute
+    function onRaiseDispute(address caller, bytes calldata data) external onlyDisputeModule nonReentrant {
+        (bytes memory claim, uint64 liveness, address currency, uint256 bond, bytes32 identifier) = abi.decode(
+            data,
+            (bytes, uint64, address, uint256, bytes32)
+        );
+
+        ArbitrationPolicyUMAStorage storage $ = _getArbitrationPolicyUMAStorage();
+        if (liveness < $.minLiveness) revert Errors.ArbitrationPolicyUMA__LivenessBelowMin();
+        if (liveness > $.maxLiveness) revert Errors.ArbitrationPolicyUMA__LivenessAboveMax();
+        if (bond > $.maxBonds[currency]) revert Errors.ArbitrationPolicyUMA__BondAboveMax();
+
+        bytes32 assertionId = OPTIMISTIC_ORACLE_V3.assertTruth(
+            claim,
+            caller, // asserter
+            address(this), // callbackRecipient
+            address(0), // escalationManager
+            liveness,
+            IERC20(currency),
+            bond,
+            identifier,
+            bytes32(0) // domainId
+        );
+
+        uint256 disputeId = IDisputeModule(DISPUTE_MODULE).disputeCounter();
+        $.assertionIdToDisputeId[assertionId] = disputeId;
+        $.disputeIdToAssertionId[disputeId] = assertionId;
+
+        emit DisputeRaisedUMA(disputeId, caller, claim, liveness, currency, bond, identifier);
+    }
+
+    /// @notice Executes custom logic on disputing judgement
+    /// @dev Enforced to be only callable by the DisputeModule
+    /// @param disputeId The dispute id
+    /// @param decision The decision of the dispute
+    /// @param data The arbitrary data used to set the dispute judgement
+    function onDisputeJudgement(uint256 disputeId, bool decision, bytes calldata data) external onlyDisputeModule {}
+
+    /// @notice Executes custom logic on disputing cancel
+    /// @dev Enforced to be only callable by the DisputeModule
+    /// @param caller Address of the caller
+    /// @param disputeId The dispute id
+    /// @param data The arbitrary data used to cancel the dispute
+    function onDisputeCancel(address caller, uint256 disputeId, bytes calldata data) external onlyDisputeModule {
+        revert Errors.ArbitrationPolicyUMA__CannotCancel();
+    }
+
+    /// @notice Executes custom logic on resolving dispute
+    /// @dev Enforced to be only callable by the DisputeModule
+    /// @param caller Address of the caller
+    /// @param disputeId The dispute id
+    /// @param data The arbitrary data used to resolve the dispute
+    function onResolveDispute(address caller, uint256 disputeId, bytes calldata data) external onlyDisputeModule {}
```

Comment:
same as above.

Reply:
addressed in https://github.com/storyprotocol/protocol-core-v1/pull/263/commits/8c4339e83561c8d124bf15104fcbf002544adc13